rust/agama-lib/src/scripts/model.rs

```diff
@@ -172,12 +312,53 @@ impl Default for ScriptsRepository {
     }
 }
 
+/// Implements the logic to run a command.
+///
+/// At this point, it only supports running a command in a chroot environment. In the future, it
+/// might implement support for other features, like progress reporting (like AutoYaST does).
+struct ScriptRunner {
+    chroot: bool,
+}
+
+impl ScriptRunner {
+    fn new() -> Self {
+        Default::default()
+    }
+
+    fn with_chroot(self, chroot: bool) -> Self {
+        Self { chroot }
```

Comment:
if we plan to have more args in future I kind of expect code like `self.chroot = chroot; self` that will just modify self so it does not loose other params.